src/ui/locationInfo/LocationInfo.tsx

```diff
@@ -1,9 +1,9 @@
-import { useLocation } from 'react-router-dom';
+import { useRouterState } from '@tanstack/react-router';
 
 import { CodeBlock } from 'ui/codeBlock/CodeBlock';
 
 export const LocationInfo = () => {
-  const location = useLocation();
+  const location = useRouterState({ select: (state) => state.location });
```

Comment:
I think it may be useful to create a custom hook `useLocation` with that logic, cause I expect it may be helpful in more places in bigger apps.